cmd/osv-scanner/testmain_test.go

```diff
@@ -1,14 +1,55 @@
 package main
 
 import (
+	"context"
+	"io"
+	"log"
+	"log/slog"
 	"os"
+	"strings"
 	"testing"
 
 	"github.com/go-git/go-git/v5"
 	"github.com/google/osv-scanner/v2/internal/testutility"
 )
 
+// muffledHandler eats certain log messages to reduce noise in the test output
+type muffledHandler struct {
+	slog.TextHandler
+}
+
+func (c *muffledHandler) Handle(ctx context.Context, record slog.Record) error {
+	if record.Level < slog.LevelError {
```

Comment:
I realised this is a good idea because as part of #1666 a log record with a level of error will trigger a non-zero exit code (effectively, that's the logic replacing the `HasErrored` path in the current `reporter` interface)